src/librustc/hir/map/collector.rs

```diff
@@ -145,7 +145,7 @@ impl<'a, 'hir> NodeCollector<'a, 'hir> {
         let mut collector = NodeCollector {
             krate,
             source_map: sess.source_map(),
-            map: repeat(None).take(sess.current_node_id_count()).collect(),
+            map: Default::default(),
```

Comment:
You can use the node count as the capacity here still.

Reply:
Actually that function is no longer available with this commit; should I bring it back or "inline" it?

Reply:
Bring it back. It was just there for this code.